### PR TITLE
perf: spawn runtime task ASAP

### PR DESCRIPTION
**Pull request by hyf0**
### Test Plan

---

### CODE REVIEW

**Comment by hyf0** [comment]
Current dependencies on/for this PR:
* main
  * **PR #275** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/275?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #276** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/276?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
      * **PR #277** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/277?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #278** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/278?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #279** <a href="https://app.graphite.dev/github/pr/rolldown-rs/rolldown/279?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/rolldown-rs/rolldown/276?utm_source=stack-comment).

File: crates/rolldown/src/bundler/stages/scan_stage.rs [hunk 1]
```diff
@@ -46,6 +46,28 @@ impl<Fs: FileSystem + Default + 'static> ScanStage<Fs> {
...
+    assert!(!self.input_options.input.is_empty(), "You must supply options.input to rolldown");
+
+    let mut module_loader = ModuleLoader::new(
+      Arc::clone(&self.input_options),
+      Arc::clone(&self.plugin_driver),
+      self.fs.share(),
+      Arc::clone(&self.resolver),
+    );
+
+    module_loader.try_spawn_runtime_module_task();
```
Comment (milesj) [inline crates/rolldown/src/bundler/stages/scan_stage.rs:59]: being called twice, not sure if intentional

Comment (hyf0) [reply crates/rolldown/src/bundler/stages/scan_stage.rs:59]: Yeah. It's designed to be abled called multiple times. The first call is to trigger the task and second is to get the runtime `ModuleId`.

https://github.com/rolldown-rs/rolldown/blob/c93a2fb3725a0ebf6f792aa9191a366ceae4fcb9/crates/rolldown/src/bundler/module_loader/module_loader.rs#L95-L106

Comment (hyf0) [reply crates/rolldown/src/bundler/stages/scan_stage.rs:59]: But calling twice is not a good sign, I have ideas to solve it, but just too many todos on the schedules.

Comment (milesj) [reply crates/rolldown/src/bundler/stages/scan_stage.rs:59]: Got it, makes sense!

**Review by underfin — APPROVED** [review]

**Comment by hyf0** [comment]
### Merge activity

* **Nov 15, 1:19 AM**: @hyf0 started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/rolldown-rs/rolldown/276).
* **Nov 15, 1:19 AM**: [Graphite](https://app.graphite.dev/github/pr/rolldown-rs/rolldown/276) rebased this pull request as part of a merge.
* **Nov 15, 1:20 AM**: @hyf0 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/rolldown-rs/rolldown/276).

**Pull request merged**